### PR TITLE
Add softfail to check package version file

### PR DESCRIPTION
**Pull request by tinawang123**
Check package version for sles12sp5, after upgrade

- Related ticket: https://progress.opensuse.org/issues/55097
- Verification run:
http://10.161.8.44/tests/519#step/check_package_version/47

### CODE REVIEW

File: tests/console/check_package_version.pm [hunk 1]
```diff
@@ -73,7 +75,11 @@ sub run {
     }
 
     assert_script_run('python3 --version', 30);
-    die "there are failed packages" if @failed_pcks;
+    record_soft_failure "poo#54257, jsc#7146 some failed packages" if @failed_pcks;
```
Comment (coolgw) [inline tests/console/check_package_version.pm]: Maybe we can store the jira number into the package struct, then we can softfail print out which Jira failed.

**Comment by tinawang123** [comment]
Verified run: http://10.161.8.44/tests/520#step/check_package_version/15

**Comment by tinawang123** [comment]
Verified run: http://10.161.8.44/tests/521#step/check_package_version/12

File: tests/console/check_package_version.pm [hunk 2]
```diff
@@ -45,35 +47,34 @@ sub cmp_version {
...
     select_console 'root-console';
 
-    my @failed_pcks;
     foreach my $key (keys %package) {
-        my $pcks = cmp_packages($key, $package{$key});
-        push @failed_pcks, $pcks if $pcks;
+        my $pcks = cmp_packages($key, $package{$key}[0], $package{$key}[1]);
     }
 
     if (get_var('ARCH') =~ /s390x/) {
```
Comment (lemon-suse) [inline tests/console/check_package_version.pm]: Please add some comments here to describe why there is special code for s390x.

File: tests/console/check_package_version.pm [hunk 3]
```diff
@@ -45,35 +47,34 @@ sub cmp_version {
...
         foreach my $key (keys %package_s390x) {
-            my $pcks = cmp_packages($key, $package_s390x{$key});
-            push @failed_pcks, $pcks if $pcks;
+            my $pcks = cmp_packages($key, $package_s390x{$key}[0], $package_s390x{$key}[1]);
         }
     }
 
     assert_script_run('python3 --version', 30);
```
Comment (lemon-suse) [inline tests/console/check_package_version.pm]: Please add Jira No. as comments or viewer will not know why run 'python3 -- version' here.

File: tests/console/check_package_version.pm [hunk 4]
```diff
@@ -45,35 +47,34 @@ sub cmp_version {
...
-    die "there are failed packages" if @failed_pcks;
+
+    zypper_call('in ebtables');
```
Comment (lemon-suse) [inline tests/console/check_package_version.pm]: Same as previous, add comments.

**Comment by coolgw** [comment]
Should base following check for phase1:
SLE-5699  & SLE-7100 & SLE-5754 : assert fix version base on Jira description
SLE-7146 & SLE-6508 : will check package whether exist after migration to 12sp5

**Comment by coolgw** [comment]
> Should base following check for phase1:
> SLE-5699 & SLE-7100 & SLE-5754 : assert fix version base on Jira description
> SLE-7146 & SLE-6508 : will check package whether exist after migration to 12sp5

New update:
SLE-7146 is rejected
SLE-7100: should check python>36 should be installed.

**Comment by tinawang123** [comment]
python36, verified run: http://10.161.8.44/tests/526#step/check_package_version/46

**Comment by tinawang123** [comment]
https://jira.suse.com/browse/SLE-5699 According the comment, openscap version is not fixed.

**Comment by coolgw** [comment]
where is SLE-5708 -QA: Add python-daemon to the distributionBLOCKED Should Have(New add 05 Aug 2019 ) python-daemon, can be checked after migration Can check python-daemon after migration
?

**Comment by tinawang123** [comment]
> where is SLE-5708 -QA: Add python-daemon to the distributionBLOCKED Should Have(New add 05 Aug 2019 ) python-daemon, can be checked after migration Can check python-daemon after migration
> ?

Added

**Comment by tinawang123** [comment]
Verified run: http://10.161.8.44/tests/539#step/check_package_version/42

**Comment by lemon-suse** [comment]
LGTM.

**Pull request merged**